src/main/java/com/intuit/graphql/orchestrator/xtext/XtextResourceSetBuilder.java

```diff
@@ -66,6 +77,10 @@ public XtextResourceSet build() {
     graphqlResourceSet = GRAPHQL_INJECTOR.getInstance(XtextResourceSet.class);
     files.forEach((fileName, content) -> {
       try {
+        if(isFederatedResourceSet) {
```

Comment:
imagine a provider having multiple files. you will be appending the FEDERATION_DIRECTIVES to each of the files, which is not ideal. We want to merge all content in the for loop and do this if outside of forloop. Also 
createGraphqlResourceFromString should be outside the for loop since you already merge everything in the loop